var/ramble/repos/builtin/workflow_managers/slurm/workflow_manager.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2022-2024 The Ramble Authors
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import os
+
+from ramble.wmkit import *
+from ramble.expander import ExpanderError
+from ramble.application import experiment_status
+
+from spack.util.executable import Executable
+
+_ensure_job_id_snippet = r"""
+job_id=$(< {experiment_run_dir}/.slurm_job)
+if [ -z "${job_id:-}" ]; then
+    echo "No valid job_id found" 1>&2
+    exit 1
+fi
+"""
+
+
+class Slurm(WorkflowManagerBase):
+    """Slurm workflow manager"""
+
+    name = "slurm"
+
+    maintainers("linsword13")
+
+    tags("workflow", "slurm")
+
+    def __init__(self, file_path):
+        super().__init__(file_path)
+
+        self.runner = SlurmRunner()
+
+    workflow_manager_variable(
+        name="partition",
+        default="",
+        description="Name of the slurm partition for job submission",
+    )
+
+    workflow_manager_variable(
+        name="job_name",
+        default="{application_name}_{workload_name}_{experiment_name}",
+        description="Slurm job name",
+    )
+
+    workflow_manager_variable(
+        name="extra_sbatch_headers",
+        default="",
+        description="Extra sbatch headers added to slurm job script",
+    )
+
```

Comment:
Can you add a `workflow_manager_variable` to define `hostlist` as `$SLURM_JOB_NODELIST` too? That might make composition with modifiers that require this to be define a bit easier.

Reply:
Great suggestion, added.